File: backend/src/main/java/com/ems/service/EmployeeService.java
package com.ems.service;

import com.ems.dto.EmployeeDTO;
import com.ems.dto.PagedResponse;
import org.springframework.data.domain.Page;

import java.util.List;

/**
 * Service interface for Employee business operations.
 *
 * Defines all core business logic operations for Employee management
 */
public interface EmployeeService {

    public EmployeeDTO saveEmployee(EmployeeDTO employeeDTO);
    PagedResponse<EmployeeDTO> getAllEmployees(String keyword, String department, int page, int size, String sortBy, String direction);
    public EmployeeDTO getEmployeeById(Long id);
    public EmployeeDTO updateEmployee(Long id, EmployeeDTO employeeDTO);
    public void deleteEmployee(Long id);
}

File: frontend/src/services/employeeService.js
import axiosInstance from "../api/axiosInstance";

export const createEmployee = async(employee) => {
    const response = await axiosInstance.post("/employees", employee);
    return response.data;
};

export const getEmployees = async ({
  keyword, department, page, size, sortBy, direction,
}) => {
    const response = await axiosInstance.get("/employees", { params: { keyword, department , page, size, sortBy, direction}
    });
    return response.data;
}

export const getEmployeeById = async (id) => {
    const response = await axiosInstance.get(`/employees/${id}`);
    return response.data;
}

export const updateEmployee = async (id, employee) => {
  const response = await axiosInstance.put(`/employees/${id}`, employee);
  return response.data;
};


export const deleteEmployee = async (id) => {
    await axiosInstance.delete(`/employees/${id}`);
};